### PR TITLE
Block import of multisite into single site and vice versa

### PR DESCRIPTION
**Pull request by markowsiak**
## Description

We should fail in the instances that a:

* multisite sql dump file was provided for a single site
* single site sql dump file was provided for a multisite

## Steps to Test

1. Check out PR.
1. Run `npm run build`
1. Execute an import that would pass -- script continues with import
1. Execute an import that would fail because the SQL dump is multi and the site is not -- scripts exits
1. Execute an import that would fail because the SQL dump is single site and the site is multisite -- script exits

Example commands coming!

---
_This PR was started by: [git wf pr](https://github.com/groupon/git-workflow/releases/tag/v1.3.2)_

### CODE REVIEW

**Comment by dchymko** [comment]
Just thinking out loud here, I wonder if we could get the multisite value first and then pass that into the validate function. It would keep the checks all in there and only scan through the file once?

File: src/bin/vip-import-sql.js [hunk 1]
```diff
@@ -35,13 +37,35 @@ const appQuery = `
...
+
+	const fileSize = await getFileSize( fileName );
+
+	if ( ! fileSize ) {
+		exit.withError( `File '${ fileName }' is empty.` );
+	}
+
+	if ( fileSize > SQL_IMPORT_FILE_SIZE_LIMIT ) {
+		exit.withError(
+			`The sql import file size (${ fileSize } bytes) exceeds the limit the limit (${ SQL_IMPORT_FILE_SIZE_LIMIT } bytes).` +
```
Comment (elazzabi) [inline src/bin/vip-import-sql.js]:
```suggestion
			`The sql import file size (${ fileSize } bytes) exceeds the limit (${ SQL_IMPORT_FILE_SIZE_LIMIT } bytes).` +
```
Fixing a small typo here

File: src/bin/vip-import-sql.js [hunk 2]
```diff
@@ -35,13 +37,35 @@ const appQuery = `
...
+				'Please split it into multiple files or contact support for assistance.' );
```
Comment (elazzabi) [inline src/bin/vip-import-sql.js:66]: > Please split it into multiple files

Is this an opportunity to provide a link to the documentation on how you can do it? Or we can just leave "Contact support for assistance" now.

File: src/lib/cli/exit.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+
+/**
+ * Internal dependencies
+ */
+
+export function withError( message: string ) {
```
Comment (elazzabi) [inline src/lib/cli/exit.js:15]: That's a clever way! I see that `exit.withError()` is much better than always calling `process.exit()`... Thanks for that! ❤️

File: src/lib/validations/line-by-line.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	} );
+}
+
+export async function getReadInterface( filename: string ) {
+	let fd;
+	try {
+		fd = await openFile( filename );
+	} catch ( e ) {
+		console.log( chalk.red( 'Error: ' ) + 'The file at the provided path is either missing or not readable.' );
+		console.log( 'Please check the input and try again.' );
```
Comment (elazzabi) [inline src/lib/validations/line-by-line.js]: Is this an opportunity to use the `exit.withError()` ?

**Review by elazzabi — DISMISSED** [review]
Left a few non-blocking comments. Otherwise, looks good to me!

File: src/bin/vip-import-sql.js (hunk 2 above)
Comment (markowsiak) [reply src/bin/vip-import-sql.js:66]: TBH I'm open to either.  This code was moved in this PR but not fabricated.  Happy to adjust as necessary.

File: src/lib/validations/line-by-line.js (hunk 4 above)
Comment (markowsiak) [reply src/lib/validations/line-by-line.js]: I will check this out, but I believe you're correct 👍

**Review by elazzabi — APPROVED** [review]
Gave this another look, it looks good 🎉

**Pull request merged**